js/coinbasepro.js

```diff
@@ -1295,15 +1295,19 @@ module.exports = class coinbasepro extends Exchange {
                 }
             }
             const what = nonce + method + request + payload;
-            const secret = this.base64ToBinary (this.secret);
-            const signature = this.hmac (this.encode (what), secret, 'sha256', 'base64');
-            headers = {
-                'CB-ACCESS-KEY': this.apiKey,
-                'CB-ACCESS-SIGN': signature,
-                'CB-ACCESS-TIMESTAMP': nonce,
-                'CB-ACCESS-PASSPHRASE': this.password,
-                'Content-Type': 'application/json',
-            };
+            try {
+                const secret = this.base64ToBinary (this.secret);
+                const signature = this.hmac (this.encode (what), secret, 'sha256', 'base64');
+                headers = {
+                    'CB-ACCESS-KEY': this.apiKey,
+                    'CB-ACCESS-SIGN': signature,
+                    'CB-ACCESS-TIMESTAMP': nonce,
+                    'CB-ACCESS-PASSPHRASE': this.password,
+                    'Content-Type': 'application/json',
+                };
+            } catch (e) {
+                throw new AuthenticationError ('Invalid secret');
```

Comment:
@Wicker25 can you elaborate a bit on this try-catch clause – when does it throw in your case? We don't have this try-catch clause in any other exchange.

Reply:
Hey, @kroitor . I just noticed the other exchanges using a base64-encoded API secret. The exception in the PR description is thrown when we pass a malformed base64 token as an API secret.

I wrapped the `base64ToBinary()` call with a try-catch clause so that we can throw a more specific exception when this happens (the issue is related to one of the parameters of the authentication and I believe that it would make sense to have an `AuthenticationError` here).

Let me know if the change makes sense to you.

Reply:
@Wicker25 in that case, i think we should probably move this edit to the base code, so that we don't duplicate it in every exchange... Although, it can be tricky to add in a uniform way across the languages. Not all users would want the extra checks yet most of them want a unified behaviour in JS/Py/PHP.

Reply:
I agree with you, @kroitor . I will try to move this check to the base code :+1:

Reply:
@Wicker25 yep, thx! The difficulty with merging it as is – is that not everyone will want the extra-check, so it has to be optional.